tests/unit/test_models.py

```diff
@@ -25,3 +28,47 @@ def test_reset_token(app, models):
         encoded_token, app.config["RSA_PRIVATE_KEY"], app.config["ALGORITHM"]
     )
     assert decoded_token["usr"] == user.id
+
+
+@pytest.mark.parametrize('input', [
+    # cookie consent
+    {
+        "type": "cookie",
+        "category": "statistics",
+        "status": "accepted"
+    },
+    # gdpr consent
+    {
+        'type': 'gdpr',
+        'category': 'newsletter',
+        'status': 'rejected',
+    }
+])
+def test_create_consent(models, input):
+    consent = Consent(**input, user=models['user'])
+    assert consent
+
+
+@pytest.mark.parametrize('input', [
+    # invalid category for cookie consent
+    {
+        "type": "cookie",
+        "category": "perfrmance",
+        "status": "accepted"
+    },
+    # invalid status (gdpr consent)
+    {
+        'type': 'gdpr',
+        'category': 'newsletter',
+        'status': 'rej',
+    },
+    # invalid status (cookie consent)
+    {
+        'type': 'cookie',
+        'category': 'statistics',
+        'status': 'akceptted',
+    }
+])
+def test_create_consent_fail(models, input):
+    with pytest.raises(Exception):
```

Comment:
Addressed in cd617de.

I'd like your opinion here. To test for correct cookie consent categories, the SQLAlchemy validators are of no help as they can validate only a single field. Instead I'm checking for it in `__init__`. However, that means that this field is checked at a different stage and raises a different exception than the fields that are checked for enum values (fields `type` and `status`). It is so because the latter two checks happen in the db and raise an exception associated with a SQL query.

Reply:
If you check on `__init__`, doesn't that mean that simply changing the attributes on an existing object wouldn't catch these errors? If so it's helpful but not solid. This could maybe be solved by defining some postgresql constraint.

Although in my opinion, I think it's most important to have rigid validation at the interfaces, i.e., the API, which you already do with the schema validator. That means that invalid state could only be introduced by erroneous code, manual interaction with the db or flask admin, which I think is acceptable (as we would rarely manually edit consents). So I would maybe consider the enum checks to be enough, and simply ignore the extra validator for the models.

Reply:
Ah, true, haven't thought about post-initialization changes. In that case, yes, I think the most sensible now is to skip the extra validation.

Reply:
Addressed in https://github.com/DD-DeCaF/iam/commit/d29b9100a4ab8f264c6ba2b4c17973f81d6837ed